src/connection/StreamSocket.php

```diff
@@ -120,4 +118,21 @@ public function disconnect()
             stream_socket_shutdown($this->stream, STREAM_SHUT_RDWR);
     }
 
+    public function setTimeout(float $timeout): void
+    {
+        parent::setTimeout($timeout);
+        $this->setTimeoutOnStream();
+    }
+
+    /**
+     * @return void
+     * @throws ConnectException
+     */
+    private function setTimeoutOnStream(): void
```

Comment:
maybe it will be nice if it have same method name like in Socket configureTimeout